lib/connection/statement.js

```diff
@@ -1570,7 +1572,11 @@ function sendSfRequest(statementContext, options, appendQueryParamOnRetry)
     // retry, update the url
     if ((numRetries > 0) && appendQueryParamOnRetry)
     {
-      options.url = Util.url.appendParam(urlOrig, 'retry', true);
+      let modifiedURL;
+      modifiedURL = Util.url.appendParam(urlOrig, 'retryCount', numRetries);
+      if(connectionConfig.getIncludeRetryContext){
+        modifiedURL = Util.url.appendParam(modifiedURL, 'retryReason', lastStatusCodeForRetry);
```

Comment:
Could you implement some tests checking retry parameters?

Reply:
I am trying to create a testing function. Actually, Util.url.appendParam is already tested in url_util.js, and I am unsure that testing the same function again for the retry is reasonable. Also, I think we do not have a way how we can test this with an integration test or an E2E test. Please let me know if you have any idea for this.

Reply:
I created a new function for this, and created a unit testing.